actors/builtin/power/power_actor.go

```diff
@@ -476,11 +473,22 @@ func (a Actor) processDeferredCronEvents(rt Runtime) {
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to flush events")
 	})
 	failedMinerCrons := make([]addr.Address, 0)
+
 	for _, event := range cronEvents {
+
+		var payload miner0.CronEventPayload
+		err := payload.UnmarshalCBOR(bytes.NewBuffer(event.CallbackPayload))
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to unmarshal miner cron payload into expected structure")
```

Comment:
Given the impl of miner actor this should never error ✅ 

Thinking more about this through the lens of paranoia I'd like to see this done a little differently. 
1) Include Payload bytes directly in the DeferredCronEventParams (renamed from DeferredCronEventPayload).  So `EventType` becomes `EventPayload` and type changes to []bytes
2) do the unmarshalling directly in the miner actor OnDeferredCronEvents and fail with a serialization exit code in the (should be impossible) case where unmarshalling fails

This way the weird contract where the miner actor sends over arbitrary bytes and power actor doesn't care what they are is logically maintained because we just send the bytes back, but now with metadata attached.  And the main benefit is that if there is a programmer error somewhere in miner or power actor causing unmarshal to fail then that failure will propagate as error exitcode from the call to miner actor below and we will maintain separation of cron failures between miner actors.

Reply:
The more correct thing to do out of scope here or for v14 is for the miner actor to send over an event type int and the power actor to store that int instead of bytes of wrapped int in the cron event queue.  

It's not obvious doing this is actually worth the effort but I filed an issue for it: https://github.com/filecoin-project/specs-actors/issues/1490

Reply:
yeehaw dooone